api/serializers/student/dashboard.py

```diff
@@ -65,3 +68,27 @@ class Meta(CommonDashboardStudentSerializer.Meta):
             "personal_info",
             "groups",
         )
+
+
+class DashboardStudentTransferSerializer(serializers.Serializer[Any]):
+    to_group_id = serializers.IntegerField()
+    from_group_id = serializers.IntegerField()
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        try:
+            to_group = Group.objects.get(pk=int(attrs["to_group_id"]))
+            attrs["to_group"] = to_group
+            if self.instance is not None and self.instance in to_group.students.all():
+                raise ConflictError("Student is already in that group")
+        except Group.DoesNotExist:
+            raise ConflictError("Transfer to group is no found")
+
+        try:
+            from_group = Group.objects.get(pk=int(attrs["from_group_id"]))
+            if self.instance is not None and self.instance not in from_group.students.all():
+                raise ConflictError("Student must be in group from which he is being transferred")
+            attrs["from_group"] = from_group
+        except Group.DoesNotExist:
+            raise ConflictError("Transfer from group is no found")
```

Comment:
definitely more of a nitpick but it would be easier to debug things if these included the ID's of things e.g `Student X must be in group Y` and so on

Reply:
Agreed!